CUDA backend compilation fails with -DBUILD_SHARED_LIBS=ON

Building the SYCL toolchain with CUDA support fails with the CMake flag `-DBUILD_SHARED_LIBS=ON`:
```
make sycl-toolchain
...
Scanning dependencies of target prepare_builtins
[ 93%] Building CXX object tools/libclc/utils/CMakeFiles/prepare_builtins.dir/prepare-builtins.cpp.o
[ 93%] Linking CXX executable ../../../bin/prepare_builtins
CMakeFiles/prepare_builtins.dir/prepare-builtins.cpp.o: In function `main':
prepare-builtins.cpp:(.text.startup.main+0x100): undefined reference to `llvm::parseBitcodeFile(llvm::MemoryBufferRef, llvm::LLVMContext&)'
prepare-builtins.cpp:(.text.startup.main+0x46c): undefined reference to `llvm::errorToErrorCodeAndEmitErrors(llvm::LLVMContext&, llvm::Error)'
collect2: error: ld returned 1 exit status
make[3]: *** [tools/libclc/utils/CMakeFiles/prepare_builtins.dir/build.make:90: bin/prepare_builtins] Error 1
make[2]: *** [CMakeFiles/Makefile2:96910: tools/libclc/utils/CMakeFiles/prepare_builtins.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:97524: tools/sycl/CMakeFiles/sycl-toolchain.dir/rule] Error 2
make: *** [Makefile:24405: sycl-toolchain] Error 2
```

The full CMake command is
```
cmake $SYCL_BASE/llvm/llvm \
  -DCMAKE_BUILD_TYPE=Release \
  -DLLVM_TARGETS_TO_BUILD="X86;PowerPC;AArch64;NVPTX" \
  -DLLVM_EXTERNAL_PROJECTS="llvm-spirv;sycl;opencl-aot" \
  -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;compiler-rt;lld;openmp;llvm-spirv;sycl;opencl-aot;libclc" \
  -DLLVM_EXTERNAL_SYCL_SOURCE_DIR=$SYCL_BASE/llvm/sycl \
  -DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR=$SYCL_BASE/llvm/llvm-spirv \
  -DLLVM_ENABLE_EH=ON \
  -DLLVM_ENABLE_PIC=ON \
  -DLLVM_ENABLE_RTTI=ON \
  -DLLVM_LIBDIR_SUFFIX=64 \
  -DBUILD_SHARED_LIBS=ON \
  -DSYCL_BUILD_PI_CUDA=ON \
  -DLIBCLC_TARGETS_TO_BUILD="nvptx64--;nvptx64--nvidiacl" \
  -DCUDA_TOOLKIT_ROOT_DIR=$CUDA_BASE
```

The compilation completes correctly with `-DBUILD_SHARED_LIBS=OFF -DSYCL_BUILD_PI_CUDA=ON`, or with `-DBUILD_SHARED_LIBS=ON -DSYCL_BUILD_PI_CUDA=OFF`, but fails if both are set.


Tested with the [`sycl`](https://github.com/intel/llvm/tree/sycl) branch as of 80b0306376bcd426b717d6ae17767c5de10b3a88 .

--- a/libclc/utils/CMakeLists.txt
+++ b/libclc/utils/CMakeLists.txt
@@ -3,6 +3,7 @@ set( LLVM_VERSION_DEFINE "-DHAVE_LLVM=0x${LLVM_VERSION_MAJOR}0${LLVM_VERSION_MIN
 # Setup prepare_builtins tools
 set(LLVM_LINK_COMPONENTS
   BitWriter
+  BitReader
   Core
   IRReader
   Support

